packages/devtools/src/scripts/executeAsyncScript.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * injector script for executeScript command
+ * According to the WebDriver spec we need to ensure to serialize all elements
+ * being returned from the script. In order to do so we attach a data attribute
+ * to them and fetch them using Puppeteer after the script execution. Instead
+ * of the element we return the dataFlag string instead to signalise that this
+ * is suppose to be an element handle
+ *
+ * @param  {HTMLElement} _            $eval fetched element
+ * @param  {String}      script       user script
+ * @param  {String}      dataProperty property name for elements being returned
+ * @param  {String}      dataFlag     flag for element
+ * @param  {Object[]}    args         user arguments for custom script
+ * @return {Object}                   result of custom script
+ */
+export default (_, script, scriptTimeout, dataProperty, dataFlag, ...commandArgs) => {
+    return new Promise((_resolve, _reject) => {
+        setTimeout(
+            () => _reject('script timeout'),
+            scriptTimeout
+        )
+
+        window.arguments = [...commandArgs, (result) => {
+            let tmpResult = result instanceof NodeList ? Array.from(result) : result
+            const isResultArray = Array.isArray(tmpResult)
+            tmpResult = isResultArray ? tmpResult : [tmpResult]
+
+            if (tmpResult.find((r) => r instanceof HTMLElement)) {
+                tmpResult = tmpResult.map((r, i) => {
+                    if (r instanceof HTMLElement) {
+                        const dataPropertyValue = `${dataFlag}_${i}`
+                        r.setAttribute(dataProperty, dataPropertyValue)
+                        return dataPropertyValue
+                    }
+
+                    return result
```

Comment:
Should this be `return r` instead?

Reply:
good question, our e2e tests are fine and return elements properly. Are you experiencing a bug?

Reply:
This intends to map a result array and change the ones that are HTMLElement, and presumably leave the others unchanged.

However, I would expect the following to fail:

```js
result = [
 HTMLDivElement,
 42,
 HTMLAnchorElement
];
```

This would become:
```js
result = [
  dataValue,
  [ HTMLDivElement, 42, HTMLAnchorElement ],
  dataValue
]
```
instead of

```js
result = [
  dataValue,
  42,
  dataValue
]
```

I'm still in the midst of upgrading to v5 and (after that) trying out devtools, so I haven't yet run into this (potential) bug. Was reviewing the changes and came across this.